azurelinuxagent/ga/monitor.py

```diff
@@ -277,6 +277,8 @@ def daemon(self):
                         MonitorHandler.HOST_PLUGIN_HEARTBEAT_PERIOD,
                         MonitorHandler.IMDS_HEARTBEAT_PERIOD).seconds
         while self.should_run:
+            # Updating the goal_state periodically for the monitor thread to keep the WireClient object upto-date always
+            self.protocol.update_goal_state()
```

Comment:
So this will update every 1m the `self.protocol` object of the monitor.py and that should not impact any protocol objects in the exthandler. Correct?

Reply:
Yes that's correct, we would have 3 different protocol objects for the 3 threads that we own. The GoalState is not being used anywhere in the environment thread so I didnt add an `update_goal_state` call in that thread. The monitor thread however needs certain data (host_plugin details and incarnation) to function properly due to which I added this update call here

Reply:
my bad... different loop... i'm not touching this on my PR, you are good